app/src/main/java/com/asfoundation/wallet/topup/TopUpFragmentPresenter.kt

```diff
@@ -158,4 +186,116 @@ class TopUpFragmentPresenter(private val view: TopUpFragmentView,
           }
         }
   }
+
+  private fun handleInvalidValueInput(packageName: String, topUpData: TopUpData) {
+    if (topUpData.currency.fiatValue != DEFAULT_VALUE) {
+      val value =
+          FiatValue(BigDecimal(topUpData.currency.fiatValue), topUpData.currency.fiatCurrencyCode,
+              topUpData.currency.fiatCurrencySymbol)
+      disposables.add(fetchMinAndMaxValues()
+          .subscribeOn(networkScheduler)
+          .observeOn(viewScheduler)
+          .doOnSuccess {
+            showValueWarning(it.first, it.second, value)
+            handleShowBonus(packageName, topUpData, it.first, it.second, value)
+          }
+          .subscribe())
+    }
+  }
+
+  private fun fetchMinAndMaxValues(): Single<Pair<FiatValue, FiatValue>> {
+    return Single.zip(interactor.getMaxTopUpValue(), interactor.getMinTopUpValue(),
+        BiFunction { maxValue: FiatValue, minValue: FiatValue -> Pair(maxValue, minValue) })
```

Comment:
I would pass the zip logic to the interactor and have the presenter call a method that would return an object with both values. (Keep both methods as private, as they can later be used as public if we only need one of the values)